pkg/tempo/config.go

```diff
@@ -246,46 +257,58 @@ type exporterConfig struct {
 }
 
 // exporter builds an OTel exporter from RemoteWriteConfig
-func exporter(remoteWriteConfig RemoteWriteConfig) (map[string]interface{}, error) {
-	if len(remoteWriteConfig.Endpoint) == 0 {
+func exporter(rwCfg RemoteWriteConfig) (map[string]interface{}, error) {
+	if len(rwCfg.Endpoint) == 0 {
 		return nil, errors.New("must have a configured a backend endpoint")
 	}
 
 	headers := map[string]string{}
-	if remoteWriteConfig.Headers != nil {
-		headers = remoteWriteConfig.Headers
+	if rwCfg.Headers != nil {
+		headers = rwCfg.Headers
 	}
 
-	if remoteWriteConfig.BasicAuth != nil {
-		password := string(remoteWriteConfig.BasicAuth.Password)
+	if rwCfg.BasicAuth != nil {
+		password := string(rwCfg.BasicAuth.Password)
 
-		if len(remoteWriteConfig.BasicAuth.PasswordFile) > 0 {
-			buff, err := ioutil.ReadFile(remoteWriteConfig.BasicAuth.PasswordFile)
+		if len(rwCfg.BasicAuth.PasswordFile) > 0 {
+			buff, err := ioutil.ReadFile(rwCfg.BasicAuth.PasswordFile)
 			if err != nil {
-				return nil, fmt.Errorf("unable to load password file %s: %w", remoteWriteConfig.BasicAuth.PasswordFile, err)
+				return nil, fmt.Errorf("unable to load password file %s: %w", rwCfg.BasicAuth.PasswordFile, err)
 			}
 			password = string(buff)
 		}
 
-		encodedAuth := base64.StdEncoding.EncodeToString([]byte(remoteWriteConfig.BasicAuth.Username + ":" + password))
+		encodedAuth := base64.StdEncoding.EncodeToString([]byte(rwCfg.BasicAuth.Username + ":" + password))
 		headers["authorization"] = "Basic " + encodedAuth
 	}
 
-	compression := remoteWriteConfig.Compression
+	compression := rwCfg.Compression
 	if compression == compressionNone {
 		compression = ""
 	}
 
 	otlpExporter := map[string]interface{}{
-		"endpoint":             remoteWriteConfig.Endpoint,
-		"compression":          compression,
-		"headers":              headers,
-		"insecure":             remoteWriteConfig.Insecure,
-		"insecure_skip_verify": remoteWriteConfig.InsecureSkipVerify,
-		"sending_queue":        remoteWriteConfig.SendingQueue,
-		"retry_on_failure":     remoteWriteConfig.RetryOnFailure,
+		"endpoint":         rwCfg.Endpoint,
+		"compression":      compression,
+		"headers":          headers,
+		"insecure":         rwCfg.Insecure,
+		"sending_queue":    rwCfg.SendingQueue,
+		"retry_on_failure": rwCfg.RetryOnFailure,
+	}
+
+	if !rwCfg.Insecure {
+		// If there is a TLSConfig use it
+		if rwCfg.TLSConfig != nil {
+			otlpExporter["ca_file"] = rwCfg.TLSConfig.CAFile
+			otlpExporter["cert_file"] = rwCfg.TLSConfig.CertFile
+			otlpExporter["key_file"] = rwCfg.TLSConfig.KeyFile
+			otlpExporter["insecure_skip_verify"] = rwCfg.TLSConfig.InsecureSkipVerify
+		} else {
+			// If not, set whatever value is specified in the old config.
+			otlpExporter["insecure_skip_verify"] = rwCfg.InsecureSkipVerify
```

Comment:
Can we do a warning log here that the field is deprecated and will be removed in a future version?

Reply:
Addressed in c274aed102e775e0ab37c48d05b7d0c76f1e6197